qutebrowser/browser/webelem.py

```diff
@@ -319,6 +319,11 @@ def _mouse_pos(self):
         else:
             rect.setHeight(rect.width())
         pos = rect.center()
+        if rect.y()<0 and (rect.y()+rect.height()-15>0):
```

Comment:
In this case, why not just click the center of the 'visible' portion of the element (so we don't have to hardcode '15')? This was a bit hard to explain so I drew an image.

![untitled](https://user-images.githubusercontent.com/4349709/38596347-fdc0f948-3d1e-11e8-9d32-14b7ad2a253f.png)

So, I think you may be able to do `rect.setY(max(0, rect.y()))`.

This will probably interact weirdly with the 'height/width' normalization on line 317. I think that putting something like this before that normalization would work, but I might have missed a corner case.

Reply:
I have thought about using the solution you mentioned, but some problem will occurred when there is overlaps. For example, in the duckduckgo image search, the menu bar is on top of the image, and sometimes the click point will be on the menu bar. 

I have come up with another solution. Because the hint will be showed on the image, we could set the click point to the the location of the hint, which will guarantee the click point is on the intended link.

Reply:
Hmm, this won' always work either because:

1. If the bar is on the botom, we will always hit the bottom of the image (meaning the image will always be unclickable)
2. If the image is thinner in height than the link, we will miss the element completely. (This is why the tests are failing, I think, some buttons in the test suite are shorter than 15px).

The proper solution to this is to use js to click the element directly (instead of sending these clicks), but I think we can keep that on hold for another PR. Using the fake-click method, there will always be cases where we don't click the correct element, but using the 'middle of visible area' method we can at least guarantee we will always hit an unobscured element properly.